test/integration/client-side-encryption/driver.test.ts

```diff
@@ -401,6 +410,227 @@ describe('Client Side Encryption Functional', function () {
+      );
+      // Load the file key1-document.json as key1Document.
+      const key1Document = EJSON.parse(
+        await fs.promises.readFile(path.join(data, 'keys', 'key1-document.json'), 'utf8'),
+        { relaxed: false }
+      );
+      // Read the "_id" field of key1Document as key1ID.
+      key1Id = key1Document._id;
+      const setupClient = this.configuration.newClient();
+      // Drop and create the collection db.explicit_encryption using encryptedFields as an option.
+      const db = setupClient.db('db');
+      await db
+        .collection('explicit_encryption')
+        .drop()
+        .catch(() => null);
+      await db.createCollection('explicit_encryption', { encryptedFields }).catch(error => {
+        const expected =
+          'Driver support of Queryable Encryption is incompatible with server'.toLowerCase();
+        if (!error.message.toLowerCase().includes(expected)) throw error;
+      });
+      // Drop and create the collection keyvault.datakeys.
+      const kdb = setupClient.db('keyvault');
+      await kdb
+        .collection('datakeys')
+        .drop()
+        .catch(() => null);
+      await kdb.createCollection('datakeys');
+      // Insert key1Document in keyvault.datakeys with majority write concern.
+      await kdb.collection('datakeys').insertOne(key1Document, { writeConcern: { w: 'majority' } });
+
+      client = this.configuration.newClient({}, { appName: 'clientEncryption' });
+      await client.connect();
+      clientEncryption = new ClientEncryption(client, {
+        kmsProviders: { local: { key: LOCAL_KEY } },
+        keyVaultNamespace,
+        keyVaultClient: null,
+        timeoutMS: 500,
+        ...getEncryptExtraOptions()
+      });
+
+      await setupClient.close();
+    });
+
+    afterEach(async function () {
+      await client?.close();
+    });
+
+    describe('rewrapManyDataKey', function () {
+      describe('when the bulk operation takes too long', function () {
+        makeBlockingFailFor('update', 2000);
+
+        it(
+          'throws a timeout error',
+          metadata,
+          runAndCheckForCSOTTimeout(async () => {
+            await clientEncryption.rewrapManyDataKey({ _id: key1Id }, { provider: 'local' });
+          })
+        );
+      });
+
+      describe('when the find operation for fetchKeys takes too long', function () {
+        makeBlockingFailFor('find', 2000);
+
+        it(
+          'throws a timeout error',
+          metadata,
+          runAndCheckForCSOTTimeout(async () => {
+            await clientEncryption.rewrapManyDataKey({ _id: key1Id }, { provider: 'local' });
+          })
+        );
+      });
+
+      // TOOD, does not make rewriteManyDataKey fail
+      describe.skip('when the find and bulk operation takes too long', function () {
+        // together they add up to 800, exceeding the timeout of 500
+        makeBlockingFailFor('update', 400);
+        makeBlockingFailFor('listCollections', 400);
```

Comment:
there can only be a single failCommand active at a time.  If you need to fail multiple commands, specify both in the failCommands array:

```typescript
        makeBlockingFailFor(['update', 'listCollections'], 400);
```

Perhaps this is why your test isn't failing?

Reply:
fixed